core/src/main/java/com/sequenceiq/cloudbreak/repository/FlowLogRepository.java

```diff
@@ -20,10 +20,17 @@
             + "AND fl.flowType != 'com.sequenceiq.cloudbreak.core.flow2.stack.termination.StackTerminationFlowConfig'")
     Set<String> findAllRunningNonTerminationFlowIdsByStackId(@Param("stackId") Long stackId);
 
-    @Query("SELECT DISTINCT fl.flowId, fl.stackId FROM FlowLog fl WHERE fl.finalized IS NULL OR fl.finalized = false")
+    @Query("SELECT DISTINCT fl.flowId, fl.stackId, fl.cloudbreakNodeId FROM FlowLog fl WHERE fl.finalized IS NULL OR fl.finalized = false")
     List<Object[]> findAllNonFinalized();
 
     @Modifying
     @Query("UPDATE FlowLog fl SET fl.finalized = true WHERE fl.flowId = :flowId")
     void finalizeByFlowId(@Param("flowId") String flowId);
+
+    @Query("SELECT fl FROM FlowLog fl WHERE fl.cloudbreakNodeId = :cloudbreakNodeId AND (fl.finalized IS NULL OR fl.finalized = false)")
+    Set<FlowLog> findAllByCloudbreakNodeId(@Param("cloudbreakNodeId") String cloudbreakNodeId);
```

Comment:
The query is looking for non finalized entries only, but the name is not match on this behavior.